Mathlib/RingTheory/TwoSidedIdeal/Lattice.lean

```diff
@@ -0,0 +1,135 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import Mathlib.RingTheory.TwoSidedIdeal.Basic
+
+/-!
+# The complete lattice structure on two-sided ideals
+-/
+
+namespace TwoSidedIdeal
+
+variable (R : Type*) [NonUnitalNonAssocRing R]
+
+instance : Sup (TwoSidedIdeal R) where
+  sup I J := { ringCon := I.ringCon ⊔ J.ringCon }
+
+lemma sup_ringCon (I J : TwoSidedIdeal R) : (I ⊔ J).ringCon = I.ringCon ⊔ J.ringCon := rfl
+
+instance : SemilatticeSup (TwoSidedIdeal R) where
+  le_sup_left I J :=  by rw [ringCon_le_iff]; exact le_sup_left
+  le_sup_right I J := by rw [ringCon_le_iff]; exact le_sup_right
+  sup_le I J K h1 h2 := by rw [ringCon_le_iff] at h1 h2 ⊢; exact sup_le h1 h2
+
+section sup
+
+variable {R}
+
+lemma mem_sup_left {I J : TwoSidedIdeal R} {x : R} :
+    x ∈ I → x ∈ I ⊔ J :=
+  fun h => (show I ≤ I ⊔ J from le_sup_left) h
+
+lemma mem_sup_right {I J : TwoSidedIdeal R} {x : R} :
+    x ∈ J → x ∈ I ⊔ J :=
+  fun h => (show J ≤ I ⊔ J from le_sup_right) h
+
+lemma mem_sup {I J : TwoSidedIdeal R} {x : R} :
+    x ∈ I ⊔ J ↔ ∃ y ∈ I, ∃ z ∈ J, y + z = x := by
+  constructor
+  · let s : TwoSidedIdeal R := .mk'
+      {x | ∃ y ∈ I, ∃ z ∈ J, y + z = x}
+      ⟨0, ⟨zero_mem _, ⟨0, ⟨zero_mem _, zero_add _⟩⟩⟩⟩
+      (by rintro _ _ ⟨x, ⟨hx, ⟨y, ⟨hy, rfl⟩⟩⟩⟩ ⟨a, ⟨ha, ⟨b, ⟨hb, rfl⟩⟩⟩⟩;
+          exact ⟨x + a, ⟨add_mem _ hx ha, ⟨y + b, ⟨add_mem _ hy hb, by abel⟩⟩⟩⟩)
+      (by rintro _ ⟨x, ⟨hx, ⟨y, ⟨hy, rfl⟩⟩⟩⟩
+          exact ⟨-x, ⟨neg_mem _ hx, ⟨-y, ⟨neg_mem _ hy, by abel⟩⟩⟩⟩)
+      (by rintro r _ ⟨x, ⟨hx, ⟨y, ⟨hy, rfl⟩⟩⟩⟩
+          exact ⟨_, ⟨mul_mem_left _ _ _ hx, ⟨_, ⟨mul_mem_left _ _ _ hy, mul_add _ _ _ |>.symm⟩⟩⟩⟩)
+      (by rintro r _ ⟨x, ⟨hx, ⟨y, ⟨hy, rfl⟩⟩⟩⟩
+          exact ⟨_, ⟨mul_mem_right _ _ _ hx, ⟨_, ⟨mul_mem_right _ _ _ hy, add_mul _ _ _ |>.symm⟩⟩⟩⟩)
+
```

Comment:
Why the blank lines?

Reply:
I though it could increase readibility as the above is the construction of some two-sided-ideal

Reply:
removed blank line